mlflow/store/tracking/file_store.py

```diff
@@ -348,17 +354,11 @@ def create_experiment(self, name, artifact_location=None, tags=None):
         self._check_root_dir()
         _validate_experiment_name(name)
         self._validate_experiment_does_not_exist(name)
-        # Get all existing experiments and find the one with largest numerical ID.
-        # len(list_all(..)) would not work when experiments are deleted.
-        experiments_ids = [
-            int(e)
-            for e in (
-                self._get_active_experiments(full_path=False)
-                + self._get_deleted_experiments(full_path=False)
-            )
-            if e.isdigit()
-        ]
-        experiment_id = max(experiments_ids) + 1 if experiments_ids else 0
+        experiment_id = (
+            _generate_unique_integer_id()
+            if self._has_experiment(FileStore.DEFAULT_EXPERIMENT_ID)
+            else FileStore.DEFAULT_EXPERIMENT_ID
+        )
```

Comment:
Can we just do:

```suggestion
        experiment_id = _generate_unique_integer_id()
```

?

Reply:
If we don't generate the default experiment when `start_run()` and it's contained `create_run()` is called within the deferred commit within autologging, the generated experiment_id from `_generate_unique_integer_id()` will not be present in the tracking store, throwing an exception. 
The original implementation also does this https://github.com/mlflow/mlflow/blob/0f062a662e846f28d1fec71137b906e905f9c4a8/mlflow/store/tracking/file_store.py#L390 with the `else 0` statement.

Reply:
(see previous CI runs in this PR where I wasn't doing this creation of the default experiment to see the test failures :) )

Reply:
Which test failures? Can you attach attach to thier links?

Reply:
This failure?
https://github.com/mlflow/mlflow/actions/runs/3261759552/jobs/5357438063#step:9:6766

Reply:
yep!

Reply:
I ran `test_autologging_disable_restores_behavior` after applying this suggestion and it passed. Am I missing something?

```diff
diff --git a/mlflow/store/tracking/file_store.py b/mlflow/store/tracking/file_store.py
index 0d3c944cd..877b943cd 100644
--- a/mlflow/store/tracking/file_store.py
+++ b/mlflow/store/tracking/file_store.py
@@ -354,11 +354,7 @@ class FileStore(AbstractStore):
         self._check_root_dir()
         _validate_experiment_name(name)
         self._validate_experiment_does_not_exist(name)
-        experiment_id = (
-            _generate_unique_integer_id()
-            if self._has_experiment(FileStore.DEFAULT_EXPERIMENT_ID)
-            else FileStore.DEFAULT_EXPERIMENT_ID
-        )
+        experiment_id = _generate_unique_integer_id()
         return self._create_experiment_with_id(name, str(experiment_id), artifact_location, tags)
 
     def _has_experiment(self, experiment_id):
```

```
> pytest tests/autologging/test_autologging_utils.py -k test_autologging_disable_restores_behavior
============================================================================================================================================== test session starts ==============================================================================================================================================
platform linux -- Python 3.7.10, pytest-7.1.3, pluggy-1.0.0 -- /home/haru/miniconda3/envs/mlflow-dev-env/bin/python
cachedir: .pytest_cache
rootdir: /home/haru/Desktop/repositories/mlflow, configfile: pytest.ini
plugins: cov-3.0.0, localserver-0.6.0
collected 68 items / 67 deselected / 1 selected                                                                                                                                                                                                                                                                 

tests/autologging/test_autologging_utils.py::test_autologging_disable_restores_behavior PASSED    
```

Reply:
@dbczumar @BenWilson2

Why `experiment_id = _generate_unique_integer_id()` didn't work: https://github.com/mlflow/mlflow/pull/7097#discussion_r998886290

Reply:
Thank you for the investigation and fix @harupy :)